### PR TITLE
Added description of results folder and results.xml

### PR DESCRIPTION
**Pull request by MatthewMWR**
Based on feedback from Intune customers, added description of results folder and results.xml, and powershell examples of working with the results.

### CODE REVIEW

File: windows/client-management/mdm/diagnosticlog-csp.md [hunk 1]
```diff
@@ -199,8 +199,107 @@ A Get to the above URI will return the results of the data gathering for the las
...
+              2 0                HKLM\Software\Microsoft\Windows\CurrentVersion\Uninstall
+              3 0                HKLM\Software\Microsoft\IntuneManagementExtension
+              4 0                HKLM\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall
+              5 0                %windir%\system32\ipconfig.exe /all
+              6 0                %windir%\system32\netsh.exe advfirewall show allprofiles
+              7 0                %windir%\system32\netsh.exe advfirewall show global
+              8 -2147024895      %windir%\system32\netsh.exe wlan show profiles
+```
+
+The next example extracts the zip archive into a customized flattened file structure. Each file name includes the directive number, HRESULT, and so on. This example could be customized to make different choices about what information to include in the file names and what formatting choices to make for special characters.
```
Comment (illfated) [inline windows/client-management/mdm/diagnosticlog-csp.md:244]: For later format editing, review and general consistency, please use default spacing between paragraphs with different content:
```suggestion
The next example extracts the zip archive into a customized flattened file structure. Each file name includes the directive number, HRESULT, and so on. This example could be customized to make different choices about what information to include in the file names and what formatting choices to make for special characters.

```

File: windows/client-management/mdm/diagnosticlog-csp.md [hunk 2]
```diff
@@ -199,8 +199,107 @@ A Get to the above URI will return the results of the data gathering for the las
 
 Each data gathering node is annotated with the HRESULT of the action and the collection is also annotated with an overall HRESULT. In this example, note that the mdmdiagnosticstool.exe command failed.
 
-The zip file which is created also contains a results.xml file whose contents align to the Data section in the SyncML for ArchiveResults. Accordingly, an IT admin using the zip file for troubleshooting can determine the order and success of each directive without needing a permanent record of the SyncML value for DiagnosticArchive/ArchiveResults.
+### Making use of the uploaded data
+The zip archive which is created and uploaded by the CSP contains a folder structure like the following:
+```
```
Comment (illfated) [inline windows/client-management/mdm/diagnosticlog-csp.md:204]: Suggestion: default spacing + syntax highlighting:
```suggestion

```powershell
```

File: windows/client-management/mdm/diagnosticlog-csp.md [hunk 3]
```diff
@@ -199,8 +199,107 @@ A Get to the above URI will return the results of the data gathering for the las
...
+Mode                 LastWriteTime         Length Name
+----                 -------------         ------ ----
+la---            1/4/2021  2:45 PM                1
+la---            1/4/2021  2:45 PM                2
+la---           12/2/2020  6:27 PM           2701 results.xml
+```
+Each data gathering directive from the original `Collection` XML corresponds to a folder in the output. For example, if the first directive was <RegistryKey HRESULT="0">HKLM\Software\Policies</RegistryKey> then folder `1` will contain the corresponding `export.reg` file.
+
+The `results.xml` file is the authoritative map to the output. It includes a status code for each directive. The order of the directives in the file corresponds to the order of the output folders. Using `results.xml` the administrator can see what data was gathered, what failures may have occurred, and which folders contain which output. For example, the following `results.xml` content indicates that registry export of HKLM\Software\Policies was successful and the data can be found in folder `1`. It also indicates that `netsh.exe wlan show profiles` command failed.
+```xml
```
Comment (illfated) [inline windows/client-management/mdm/diagnosticlog-csp.md:218]: Suggestion: default spacing between different content paragraphs:
```suggestion
The `results.xml` file is the authoritative map to the output. It includes a status code for each directive. The order of the directives in the file corresponds to the order of the output folders. Using `results.xml` the administrator can see what data was gathered, what failures may have occurred, and which folders contain which output. For example, the following `results.xml` content indicates that registry export of HKLM\Software\Policies was successful and the data can be found in folder `1`. It also indicates that `netsh.exe wlan show profiles` command failed.

```xml
```

File: windows/client-management/mdm/diagnosticlog-csp.md [hunk 4]
```diff
@@ -199,8 +199,107 @@ A Get to the above URI will return the results of the data gathering for the las
...
+    {
+        $leafSummaryString = $directiveSummaryString,$file.Name -join ' '
+        Copy-Item $file.FullName -Destination (Join-Path -Path $reformattedArchivePath -ChildPath $leafSummaryString)
+    }
+}
+#endregion 
+Remove-Item -Path $diagnosticArchiveTempUnzippedPath -Force -Recurse
+```
+That example script produces a set of files similar to the following, which can be a useful view for an administrator interactively browsing the results without needing to navigate any sub-folders or refer to `results.xml` repeatedly. 
+```
```
Comment (illfated) [inline windows/client-management/mdm/diagnosticlog-csp.md:287]: Suggestion: default spacing + syntax highlighting:
```suggestion
That example script produces a set of files similar to the following, which can be a useful view for an administrator interactively browsing the results without needing to navigate any sub-folders or refer to `results.xml` repeatedly. 

```powershell
```

**Review by illfated — APPROVED** [review]
Fair enough, looks good to me.

**Comment by e0i** [comment]
cc @JohanFreelancer9

File: windows/client-management/mdm/diagnosticlog-csp.md [hunk 5]
```diff
@@ -199,8 +199,111 @@ A Get to the above URI will return the results of the data gathering for the las
...
+
+```xml
+<Collection HRESULT="0">
+    <ID>268b3056-8c15-47c6-a1bd-4bc257aef7b2</ID>
+    <RegistryKey HRESULT="0">HKLM\Software\Policies</RegistryKey>
+    <Command HRESULT="-2147024895">%windir%\system32\netsh.exe wlan show profiles</Command>
+</Collection>
+```
+
+Administrators can apply automation to 'results.xml' to create their own preferred views of the data. For example, the following PowerShell one-liner extracts from the xml an ordered list of the directives with status code and details.
```
Comment (JohanFreelancer9) [inline windows/client-management/mdm/diagnosticlog-csp.md:228]:
```suggestion
Administrators can apply automation to 'results.xml' to create their own preferred views of the data. For example, the following PowerShell one-liner extracts from the XML an ordered list of the directives with status code and details.
```

File: windows/client-management/mdm/diagnosticlog-csp.md [hunk 6]
```diff
@@ -199,8 +199,111 @@ A Get to the above URI will return the results of the data gathering for the las
...
+```powershell
+Select-XML -Path results.xml -XPath '//RegistryKey | //Command | //Events | //FoldersFiles' | Foreach-Object -Begin {$i=1} -Process { [pscustomobject]@{DirectiveNumber=$i; DirectiveHRESULT=$_.Node.HRESULT; DirectiveInput=$_.Node.('#text')} ; $i++}
+```
+This example produces output similar to the following.
```
Comment (JohanFreelancer9) [inline windows/client-management/mdm/diagnosticlog-csp.md:232]:
```suggestion
This example produces output similar to the following:
```

File: windows/client-management/mdm/diagnosticlog-csp.md [hunk 7]
```diff
@@ -199,8 +199,111 @@ A Get to the above URI will return the results of the data gathering for the las
...
+    foreach( $file in $directiveOutputFiles)
+    {
+        $leafSummaryString = $directiveSummaryString,$file.Name -join ' '
+        Copy-Item $file.FullName -Destination (Join-Path -Path $reformattedArchivePath -ChildPath $leafSummaryString)
+    }
+}
+#endregion 
+Remove-Item -Path $diagnosticArchiveTempUnzippedPath -Force -Recurse
+```
+That example script produces a set of files similar to the following, which can be a useful view for an administrator interactively browsing the results without needing to navigate any sub-folders or refer to `results.xml` repeatedly.
```
Comment (JohanFreelancer9) [inline windows/client-management/mdm/diagnosticlog-csp.md:289]:
```suggestion
That example script produces a set of files similar to the following, which can be a useful view for an administrator interactively browsing the results without needing to navigate any sub-folders or refer to `results.xml` repeatedly:
```

**Review by JohanFreelancer9 — CHANGES_REQUESTED** [review]
Copy review - minor edits.

File: windows/client-management/mdm/diagnosticlog-csp.md (hunk 5 above)
Comment (illfated) [reply windows/client-management/mdm/diagnosticlog-csp.md:228]: Right. I forgot to notice the incorrect casing for 3 letter acronyms / initialisms. Good point.

File: windows/client-management/mdm/diagnosticlog-csp.md (hunk 6 above)
Comment (illfated) [reply windows/client-management/mdm/diagnosticlog-csp.md:232]: I agree. Good point.

File: windows/client-management/mdm/diagnosticlog-csp.md (hunk 7 above)
Comment (illfated) [reply windows/client-management/mdm/diagnosticlog-csp.md:289]: Fair point. I agree here as well.

**Comment by MatthewMWR** [comment]
Hi all. Sorry I am unfamiliar with the process. What do I do next? Do I take some action to incorporate the suggested changes and submit a new PR--- or maybe the suggested changes are automatically applied?

File: windows/client-management/mdm/diagnosticlog-csp.md (hunk 5 above)
Comment (MatthewMWR) [reply windows/client-management/mdm/diagnosticlog-csp.md:228]: Sorry am unfamiliar with the process. Am I meant to click on "Resolve conversation" ?

**Review by MatthewMWR — COMMENTED** [review]
I clicked Commit to accept the various suggestions. I hope that was the right thing to do. Thanks.

**Comment by illfated** [comment]
@MatthewMWR : That seems to be quite correct to me. In the future, whenever you are in doubt, feel free to ask a question in the next comment box below the suggestion, in case you would like to amend or adjust the suggested change. Moreover, the suggestions from @JohanFreelancer9 are usually quite correct, unless you find some kind of deviation from what you find to be the most accurate description in the related paragraph or sentence. So far, so good. Thank you.

**Review by illfated — APPROVED** [review]
Review viewing options:

- [ ] **Regular source view:** https://github.com/MicrosoftDocs/windows-itpro-docs/pull/8909/files
- [ ] **Hide whitespace changes:** https://github.com/MicrosoftDocs/windows-itpro-docs/pull/8909/files?diff=unified&w=1 (might be less useful, since most of the change consists of added paragraphs and continuous sections)
- [X] **Rich Diff view:** https://github.com/MicrosoftDocs/windows-itpro-docs/pull/8909/files?diff=unified&w=1&short_path=d255792#diff-d255792227dec1d06707c3c6af2acbe4166ca73d9f0ceb1a11c0c0f490c3f32b (publishing format preview)

**Review by JohanFreelancer9 — APPROVED** [review]

**Comment by illfated** [comment]
#sign-off

**Comment by MatthewMWR** [comment]
Is this ready to go live on docs.microsoft.com? Do I need to do anything?

**Comment by illfated** [comment]
No changes requested, this is confirmed "good to go" from our side (JohanFreelancer9 and me, us reviewers). We are simply waiting for the final review & merge from the document author (@ManikaDhiman), or the team manager Daniel Simpson, or anyone on the MS Docs team with merging rights & deciding rights to review & merge.

**Comment by illfated** [comment]
@e0i (for your information, in case I have overstepped my user boundaries)

**Comment by Dansimp** [comment]
I will need to review with the product team. More soon.

**Comment by MatthewMWR** [comment]
Thanks Dan. I was/am the feature PM for DiagnosticArchive. This current editing effort is focused on filling a gap in the original docs.

**Comment by Dansimp** [comment]
Great! Thanks for your contributions here, Matthew. Merging...

**Pull request merged**